components/style/gecko_string_cache/mod.rs

```diff
@@ -358,3 +340,21 @@ impl From<*mut nsIAtom> for Atom {
         }
     }
 }
+
+impl From<already_AddRefed<nsIAtom>> for Atom {
+    #[inline]
+    fn from(ptr: already_AddRefed<nsIAtom>) -> Atom {
+        let ptr = ptr.take();
+        debug_assert!(!ptr.is_null());
+        unsafe { Atom(WeakAtom::new(ptr)) }
+    }
+}
```

Comment:
Hmmm, `already_AddRefed` itself is nullable in Gecko code... I guess we should probably just make it non-nullable in Rust side.